step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java

```diff
@@ -227,14 +227,37 @@ public Integer getAutoNumberOfThreads() {
 	}
 
 	protected boolean isAutoNumberOfThreadsConsumed() {
-		Object autoNumberOfThreads = executionContext.getVariablesManager().getVariableAsString(EXECUTION_THREADS_AUTO_CONSUMED, "false");
+		Object autoNumberOfThreads = executionContext.getVariablesManager().getVariableAsString(EXECUTION_THREADS_AUTO_CONSUMED, Boolean.FALSE.toString());
 		return Boolean.parseBoolean(autoNumberOfThreads.toString());
 	}
 
 	protected void consumeAutoNumberOfThreads() {
-		executionContext.getVariablesManager().putVariable(executionContext.getCurrentReportNode(), EXECUTION_THREADS_AUTO_CONSUMED, "true");
+		executionContext.getVariablesManager().putVariable(executionContext.getCurrentReportNode(), EXECUTION_THREADS_AUTO_CONSUMED, Boolean.TRUE.toString());
 	}
-	
+
+	public Integer forecastNumberOfThreads(Integer originalNumberOfThreads) {
+		return forecastNumberOfThreads(originalNumberOfThreads, false);
+	}
+
+	public Integer forecastNumberOfThreads(Integer originalNumberOfThreads, boolean originalAsMaximum) {
+		Integer execution_threads_auto = getAutoNumberOfThreads();

```

Comment:
Why using "_" in the variable name? seems quite uncommon in java...

Reply:
Ehehe... yes, I know. This name directly corresponds to the variable name that the user sets, so I reused it in code to (potentially, if one knows the logic) highlight the relationship. And I just couldn't think of a better one :D